packages/cli/src/util/get-latest-version/index.ts

```diff
@@ -0,0 +1,142 @@
+import semver from 'semver';
+import XDGAppPaths from 'xdg-app-paths';
+import { dirname, parse as parsePath, resolve as resolvePath } from 'path';
+import type { Output } from '../output';
+import { existsSync, outputJSONSync, readJSONSync } from 'fs-extra';
+import type { PackageJson } from '@vercel/build-utils';
+import { spawn } from 'child_process';
+
+interface GetLatestVersionOptions {
+  cacheDir?: string;
+  distTag?: string;
+  getLatestInterval?: number;
+  output?: Output;
+  pkg: PackageJson;
+}
+
+interface PackageInfoCache {
+  version: string;
+  expireAt: number;
+  notified: boolean;
+}
+
+interface GetLatestWorkerPayload {
+  cacheFile?: string;
+  distTag?: string;
+  getLatestInterval?: number;
+  name?: string;
+}
+
+/**
+ * Determines if it needs to check for a newer CLI version and returns the last
+ * detected version. The version could be stale, but still newer than the
+ * current version.
+ *
+ * @returns {String|undefined} If a newer version is found, then the lastest
+ * version, otherwise `undefined`.
+ */
+export default function getLatestVersion({
+  cacheDir = XDGAppPaths('com.vercel.cli').cache(),
+  distTag = 'latest',
+  output,
+  pkg,
+  getLatestInterval,
```

Comment:
What is the purpose of `getLatestInterval`?

Its not being passed in by the `src/index.ts` caller.

Reply:
Just an option. If it's not beneficial, it can be removed in favor of a constant.